pkg/common/peertracker/uds_bsd.go

```diff
@@ -3,6 +3,9 @@
 package peertracker
```

Comment:
The peertracker package is all about accepting connections from a colocated process, and ensuring that the thing on the other end of the connection is always the same thing (and that the connection does not get handed off). Since these additions are not directly related to that task, I'd much prefer to see this logic live somewhere else.

For example, we don't know that the process package operations for UID and GID lookup here are safe from PID race. Instead, we want peertracker to handle the PID race problem, and do the UID and GID lookups separately while we are being protected by peertracker e.g.:
```
watcher, _ := peertracker.WatcherFromContext(ctx)

uid, gid, _ := doUidGidLookup(watcher.PID())

if !watcher.IsAlive() {
    // Something is wrong
}
```